src/components/Shops/Shops.jsx

```diff
@@ -124,6 +131,11 @@ const Shops = () => {
     scrollToTop();
   };
 
+  // Handle search
+  const handleSearchChange = event => {
+    setQuery(event.target.value);
```

Comment:
This needs throttling.

Reply:
have debounced instead of throttled, too 100ms. using lodash for this (don't worry, i'm only importing lodash debounce which only adds 1.4k gzipped)